stance/predict/overlay.py

```diff
@@ -13,7 +16,51 @@ def display_overlay(input_video, template_skeletons):
         path to the video of the user performing the motion
     template_skeletons : List[Skeletons]
     """
-    pass
+    cap = cv2.VideoCapture(input_video)
+    numFrames = int(cap.get(cv2.CAP_PROP_FRAME_COUNT))
+    width = int(cap.get(cv2.CAP_PROP_FRAME_WIDTH))
+    height = int(cap.get(cv2.CAP_PROP_FRAME_HEIGHT))
+
+    # used for scaling template skeleton to user video
+    scale_x = width / 1280
+    scale_y = height / 720
+    count = 0
+
+    for i in range(numFrames):
+        _, frame = cap.read()
+
+        # User skeleton
+        temp = list(user_skeletons[i].vectors.values())
+        for points in temp:
+            if points is not None:
+                pointA = (points.head[0], points.head[1])
+                pointB = (points.tail[0], points.tail[1])
+                cv2.line(frame, pointA, pointB, (0, 255, 255), 4)
+
+        # Template skeleton (scale and translate to align with center of gravity)
+        if count < len(benchmark_indices) and i == benchmark_indices[count]:
+            temp_skeleton = template_skeletons[count]
+            temp = list(temp_skeleton.vectors.values())
+            count += 1
+            user_center = user_skeletons[i].vectors['l_spine'].head
+            temp_center = temp_skeleton.vectors['l_spine'].head
+            dif_x = temp_center[0] * scale_x - user_center[0]
+            dif_y = temp_center[1] * scale_y - user_center[1]
+            for points in temp:
+                if points is not None:
+                    pointA = (int(points.head[0] * scale_x - dif_x), int(points.head[1] * scale_y - dif_y))
+                    pointB = (int(points.tail[0] * scale_x - dif_x), int(points.tail[1] * scale_y - dif_y))
+                    cv2.line(frame, pointA, pointB, (0, 255, 0), 4)
+        if frame is not None:
+            cv2.imshow('Frame', frame)
+        # pause video when skeleton is visible
+        if i == benchmark_indices[count - 1] + 1:
+            time.sleep(1)
+        if cv2.waitKey(70) & 0xFF == ord('q'):
+            break
+
+    cap.release()
+    cv2.destroyAllWindows()
 
 
 def compare_user_skeletons(input_video, idxs, user_skeletons, motion):
```

Comment:
I know its not part of this PR, but is this code deprecated @aspadiyath ?

Reply:
No, it does a different task. I just put it in overlay for simplicity.